### DISCUSSION TITLE
rust-course/关于生命周期和特征对象的几个问题

### DISCUSSION THREAD

**Discussion by JesseAtSZ** [sunface/rust-course#296] [内容建议]
深入生命周期
71行，这里的因果关系貌似颠倒了？应该是返回值的生命周期由入参的生命周期决定。
125行，没读懂无解生命周期与普通生命周期的区别
160行，为什么一个类型存在生命周期，我理解生命周期只存在于引用中，所以这里拿一个类型和一个引用的生命周期比较，有点不太理解

认识生命周期
517行那一段觉得还可以展开讲一下，“引用”、“被引用”这里有一点点读不懂

特征对象
特征对象的限制这个部分感觉讲得不是很清楚，294行那一段

**Comment by sunface** [comment]
好。。。还挺多的，我仿佛看到了长路漫漫🤣
